src/hermes/config.py

```diff
@@ -4,78 +4,117 @@
+    'handlers': {
+        'terminal': {
+            'class': "logging.StreamHandler",
+            'formatter': "plain",
+            'level': "INFO",
+            'stream': "ext://sys.stdout",
         },
 
-        'handlers': {
-            'terminal': {
-                'class': "logging.StreamHandler",
-                'formatter': "plain",
-                'level': "INFO",
-                'stream': "ext://sys.stdout",
-            },
-
-            'logfile': {
-                'class': "logging.FileHandler",
-                'formatter': "logfile",
-                'level': "DEBUG",
-                'filename': "hermes.log",
-            },
-
-            'auditfile': {
-                'class': "logging.FileHandler",
-                'formatter': "plain",
-                'level': "DEBUG",
-                'filename': "hermes-audit.md",
-                'mode': "w",
-            },
+        'logfile': {
+            'class': "logging.FileHandler",
+            'formatter': "logfile",
+            'level': "DEBUG",
+            'filename': "hermes.log",
         },
 
-        'loggers': {
-            'cli': {'level': "DEBUG", 'handlers': ['terminal']},
-            'hermes': {'level': "DEBUG", 'handlers': ['terminal', 'logfile']},
-            'audit': {'level': "DEBUG", 'handlers': ['terminal', 'logfile']},
+        'auditfile': {
+            'class': "logging.FileHandler",
+            'formatter': "plain",
+            'level': "DEBUG",
+            'filename': "hermes-audit.md",
+            'mode': "w",
         },
     },
+
+    'loggers': {
+        'cli': {'level': "DEBUG", 'handlers': ['terminal']},
+        'hermes': {'level': "DEBUG", 'handlers': ['terminal', 'logfile']},
+        'audit': {'level': "DEBUG", 'handlers': ['terminal', 'logfile']},
+    },
 }
 
+# This dict caches all the different configuration sections already loaded
+_config = {
+    # We need some basic logging configuration to get logging up and running at all
+    'logging': _logging_config,
+}
+
+
+def configure(config_path: pathlib.Path):
+    """
+    Load the configuration from the given path as global hermes configuration.
 
-def configure():
-    if 'hermes' in _config:
+    :param config_path: The path to a TOML file containing HERMES configuration.
+    """
+    if 'hermes' in _config and _config['hermes']:
         return
 
     # Load configuration if not present
     try:
-        with open('pyproject.toml', 'r') as config_file:
-            config_toml = toml.load(config_file)
-            hermes_config = config_toml['tool'].get('hermes', {})
+        with open(config_path, 'r') as config_file:
+            hermes_config = toml.load(config_file)
             _config['hermes'] = hermes_config
             _config['logging'] = hermes_config.get('logging', _config['logging'])
 
-    except IOError:
-        pass
+    # Exceptions are currently handled gracefully. Is this okay?
```

Comment:
I don't think this is a good idea. If the configuration file is not found, this means the user intended to configure something but messed up. In this case, hermes using its default behavior is probably the one thing they tried to avoid.

In the case where no file name is given and the default file is not found, I'm not sure what to do. Would it make sense to run hermes without any configuration? Probably for harvesting and such. For depositing data, it might not be possible (depending on how we set the default site for deposits). But the user also could have intended to create a file with the default name but accidentally misspelled it.